src/compas/geometry/brep/brep.py

```diff
@@ -0,0 +1,410 @@
+        raise NotImplementedError
+
+    @property
+    def faces(self):
+        raise NotImplementedError
+
+    @property
+    def shells(self):
+        raise NotImplementedError
+
+    @property
+    def solids(self):
+        raise NotImplementedError
+
+    # ==============================================================================
+    # Geometric Properties
+    # ==============================================================================
+
+    @property
+    def frame(self):
+        raise NotImplementedError
+
+    @property
+    def area(self):
+        raise NotImplementedError
+
+    @property
+    def volume(self):
+        raise NotImplementedError
+
+    @property
+    def centroid(self):
+        raise NotImplementedError
+
+    # ==============================================================================
+    # Constructors
+    # ==============================================================================
+
+    @classmethod
+    def from_shape(cls, shape):
+        raise NotImplementedError
+
+    @classmethod
+    def from_step(cls, filename):
+        raise NotImplementedError
+
+    @classmethod
+    def from_polygons(cls, polygons):
+        raise NotImplementedError
+
+    @classmethod
+    def from_curves(cls, curves):
+        raise NotImplementedError
+
+    @classmethod
+    def from_box(cls, box):
+        return new_brep_from_box(cls, box)
+
+    @classmethod
+    def from_sphere(cls, sphere):
+        raise NotImplementedError
+
+    @classmethod
+    def from_cylinder(cls, cylinder):
+        raise NotImplementedError
+
+    @classmethod
+    def from_cone(cls, cone):
+        raise NotImplementedError
+
+    @classmethod
+    def from_torus(cls, torus):
+        raise NotImplementedError
+
+    @classmethod
+    def from_mesh(cls, mesh):
+        raise NotImplementedError
+
+    @classmethod
+    def from_faces(cls, faces):
+        raise NotImplementedError
+
+    @classmethod
+    def from_extrusion(cls, curve, vector):
+        pass
+
+    @classmethod
+    def from_sweep(cls, profile, path):
+        pass
+
+    # create pipe
+    # create patch
+    # create offset
+
+    # ==============================================================================
+    # Boolean Constructors
+    # ==============================================================================
+
+    @classmethod
+    def from_boolean_difference(cls, A, B):
+        raise NotImplementedError
+
+    @classmethod
+    def from_boolean_intersection(cls, A, B):
+        raise NotImplementedError
+
+    @classmethod
+    def from_boolean_union(cls, A, B):
+        raise NotImplementedError
+
+    # ==============================================================================
+    # Converters
+    # ==============================================================================
+
+    def to_json(self, filepath):
+        raise NotImplementedError
+
+    def to_step(self, filepath, schema="AP203", unit="MM"):
```

Comment:
`"AP203"` and `"MM"` should go somewhere up as constants. Is it ok / consistent with the rest of COMPAS to default to mm units instead of meters?

Reply:
should be removed anyway, because in my experience, using these options results in failed step conversions. their usage is commented out in the occ implementation, but i had not yet removed the params...